packages/api/endpoints/granules.js

```diff
@@ -140,37 +142,48 @@ async function del(req, res) {
   log.info(`granules.del ${granuleId}`);
 
   const granuleModelClient = new models.Granule();
+  const granulePgModel = new GranulePgModel();
+  const filePgModel = new FilePgModel();
 
-  let granule;
+  const knex = await getKnexClient({ env: process.env });
+
+  let dynamoGranule;
+  let pgGranule;
   try {
-    granule = await granuleModelClient.getRecord({ granuleId });
+    dynamoGranule = await granuleModelClient.getRecord({ granuleId });
+    pgGranule = await granulePgModel.get(knex, { granule_id: granuleId });
```

Comment:
What about a situation where someone is attempting to delete a granule that hasn't been migrated yet?

Reply:
Hmm yeah. Right now this logic is:

"If a granule doesn't exist in **both** Dynamo **and** PG, throw an error"

Bur you're right, that doesn't seem like what we want. There's your scenario or what happens after they've completely switched from Dynamo to PG and stopped the parallel writes? I guess that's a whole separate issue but this Delete would fail in that situation. Maybe it should be:

"If a granule doesn't exist in **either** Dynamo **or** PG, throw an error".

If we do that they could have a record in Dynamo that hasn't been migrated to PG and we'll still delete the Dynamo record. Likewise, they could have a PG record and no Dynamo record and we'll still delete the PG record.

Reply:
I'm kind of wondering why we do a `get` in our `delete` handler. Is the only reason to be able to determine if the granule is `published` and throw an error if so?

If that is the case, I'd prefer that we shift that logic for delete conditions into the model code doing deletes by adding a WHERE published != true for PG and using the equivalent conditional expression for Dynamo. The other benefit of this change is that it would not be subject to race conditions because the conditions for delete would be evaluated at query time. Right now, the granule is read out of the DB and then checked before delete is attempted, but in theory a granule could be updated between when it's read and we try to delete it.

However, I'd say this effort is probably out of scope for this PR, since your changes seems consistent with existing behavior

Reply:
I think this still needs resolved.  Sorry the reply took so long -- 

@markdboyd  should weigh in too, but I think what we want is "if the granule doesn't exist in Dynamo, throw an error", as we're *explicitly reading from Dynamo* for Phase 1.     If it doesn't exist in postgres, *shrug*, as long as the postgres query is setup such that it doesn't throw an error in either case. 

We're going to have to refactor this for phase 2, at which point the logic is much simpler.

Reply:
> I think what we want is "if the granule doesn't exist in Dynamo, throw an error", as we're explicitly reading from Dynamo for Phase 1. If it doesn't exist in postgres, shrug, as long as the postgres query is setup such that it doesn't throw an error in either case.

This makes sense to me
We're going to have to refactor this for phase 2, at which point the logic is much simpler.
@markdboyd

Reply:
Updated this logic and added a few clarifying comments here:

https://github.com/nasa/cumulus/pull/2057/commits/4f1ff53d0bfc696c3df9ed3c4c6b50fbaf617e2a#diff-624123a7dc5bb5513e3769b27fe85821577087609a9376829676b7edbeb3a5aeR171-R173

Reply:
Oh in-line comments don't appear when you link to a commit like that... lame. they're available at the bottom of this Conversation page.